neffint/fixed_grid_fourier_integral.py

```diff
@@ -0,0 +1,247 @@
+
+            # Check for convergence
+            min_psi_component = np.min([np.abs(np.real(psi[taylor_mask])), np.abs(np.imag(psi[taylor_mask]))], axis=0)
+            psi_remaining_terms_bound = abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+4)*(n+5))
+            psi_converged = np.all(psi_remaining_terms_bound < rel_tolerance*min_psi_component)
+        
+        # Update variables
+        jx_to_n *= 1j*xx
+        abs_x_to_n_plus_1 *= np.abs(xx)
+        inv_factorial /= n+1
+
+    # If loop finished without converging, raise error
+    if not (phi_converged and psi_converged):
+        raise RuntimeError(f"Phi and Psi calculation failed to converge after {MAX_PHI_PSI_ITERATIONS} iterations")
+    
+    return phi, psi
+
+def fourier_integral_inf_correction(times: np.ndarray, omega_end: float, func_value_end: ArrayLike, func_derivative_end: ArrayLike=0.):
+    """Calculate the asymptotic correction term as omega->inf of a Fourier integral for the given times.
+    Uses a first or second (if func_derivative_end is given) order Taylor expansion around the angular frequency omega_end.
+
+    :param times: A 1D array of length M with the times to compute the Fourier integral for
+    :type times: np.ndarray
+    :param omega_end: A frequency to expand the Taylor series from
+    :type omega_end: float
+    :param func_value_end: An array of shape (X1, X2, ...) containing the values of the function to be transformed evaluated at omega_end
+    :type func_value_end: ArrayLike
+    :param func_derivative_end: An array of shape (X1, X2, ...) containing the derivative of the function to be transformed, evaluated at omega_end, defaults to 0.
+    :type func_derivative_end: ArrayLike, optional
+    :return: The asymptotic terms of the Fourier Integral for all times, given as an array of shape (M, X1, X2, ...)
+    :rtype: np.ndarray
+    """
+
+    # TODO: Reconsider func_value_end and func_derivative_end types, shapes and type hints
+    
+    func_value_end = np.asarray(func_value_end)
+    func_derivative_end = np.asarray(func_derivative_end)
+    
+    # Add axes from function output
+    for _ in range(func_value_end.ndim):
+        times = times[..., np.newaxis]
+    
+    # Add time axis
+    func_value_end = func_value_end[np.newaxis, ...]
+    func_derivative_end = func_derivative_end[np.newaxis, ...]
+    
+    return np.exp(1j*times*omega_end) * (1j*func_value_end/times - func_derivative_end/times**2)
+
+def fourier_integral_fixed_sampling_pchip(
+    times: ArrayLike,
+    frequencies: ArrayLike,
+    func_values: ArrayLike,
+    inf_correction_term: bool
+) -> np.ndarray:
+    """Calculates the fourier integral of the function, given as an array of values corresponding to an array of frequencies, for the time values given as input.
+    A Filon type algorithm using a piecewise cubic Hermite interpolating polynomial (pchip) and optionally an asymptotic correction term.
+    For details on implementation, see [1].
```

Comment:
Probably the name and/or the docstring should mention that
1) It is unnormalized
2) The integration factor is $e^{i\omega t}$, as opposed to $e^{-i\omega t}$
3) It is intended to do the integral over half the domain, i.e. $[0, \infty)$ and not $(-\infty, \infty)$ (What is a concise way of phrasing this?)

The third point I'm not sure about, it might work for the full domain, haven't thought much about it.